util/dvsim/NcLauncher.py

```diff
@@ -86,49 +85,49 @@ def _do_launch(self):
 
         self._dump_env_vars(exports)
 
+        # using os.open instead of fopen as this allows
+        # sharing of file descriptors across processes
+        fd = os.open(self.deploy.get_log_path(), os.O_WRONLY | os.O_CREAT)
```

Comment:
I'm not sure it really matters, but would it be easier to use `open` directly (with `closefd=False`, I think) and then use the file object's `fileno()` method to get the fd?

Reply:
I tried this, but the open command failed with this error message
Cannot use closefd=False with file name
Hence leaving the code as it is